cgrok/predicates.c

```diff
@@ -269,8 +271,9 @@ int grok_predicate_strcompare_init(grok_t *grok, grok_capture *gct,
   gpst->value = malloc(gpst->len);
   memcpy(gpst->value, args + pos, gpst->len);
 
-  gct->predicate_func_name = strdup("grok_predicate_strcompare");
-  gct->predicate_func_name_len = strlen("grok_predicate_strcompare");
+  size_t constlen = strlen("grok_predicate_strcompare");
+  gct->predicate_func_name = string_ndup("grok_predicate_strcompare", constlen);
+  gct->predicate_func_name_len = constlen;
```

Comment:
Could also do `strlen` in the <a href="http://stackoverflow.com/questions/5022067/strlen-in-the-c-preprocessor">C pre-processor</a>.